numba/core/withcontexts.py

```diff
@@ -127,29 +127,38 @@ class _ObjModeContextType(WithContext):
 
     Use this as a function that takes keyword arguments only.
     The argument names must correspond to the output variables from the
-    with-block.  Their respective values are strings representing the expected
-    types.  When exiting the with-context, the output variables are cast
+    with-block.  Their respective values can be:
+
+    1. strings representing the expected types; i.e. ``"float32"``.
+    2. global or closure variables referring to the expected type.
```

Comment:
```suggestion
    2. global or free variables referring to the expected type.
```

Reply:
I especially want to mention that the "closure variables" must be bounded at compile time.

Reply:
So it might have to be "compile-time bounded global or nonlocal variables ..."

Reply:
Ah yes, from the code, `free` is too much, the latter sentence suggestion looks good.